backends/dpdk/dpdkHelpers.cpp

```diff
@@ -1123,6 +1123,12 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
             }
             auto action = a->expr->arguments->at(0)->expression;
             auto action_name = action->to<IR::StringLiteral>()->value;
+            auto it = structure->defActs.find(action_name);
+            if (it != structure->defActs.end()) {
+                  ::error(ErrorType::ERR_UNEXPECTED, "%1%: action cannot be default action: %2%:",
```

Comment:
Similarly for this error message.

Reply:
A new P4 program is added by this PR to test above code.  See p4c/testdata/p4_16_samples/pna-example-miss.p4

Reply:
I see that test program.  I also see the expected output files with suffixes .p4-stderr and .p4-error are both empty in this PR.  I don't see how this test is passing if the error message is actually printed.

Perhaps consider adding the test programs that should always print compile-time errors in the p4_16_errors and p4_16_errors_outputs directory, instead of the 'samples' directories?  I believe that is their intent.

Reply:
In fact, the automated tests are failing for this test program, probably because the .p4-error and/or .p4-stderr files are empty, instead of containing the expected error messages.  You can run the test script under the `build` directory with the `-f` command line option to cause the actual output files to be copied into the p4_16_samples_outputs directory of your local cloned copy of this repo, where you are developing these changes.

Reply:
I have used -f for a while with p4c. For some reason -f  is not writing output. Will try errors directory one more time.

```p4
$ dpdk/testdata/p4_16_samples/pna-example-addhit.p4.test -f
./p4c-dpdk --dump ./tmpqfb7g5hb -o ./tmpqfb7g5hb/pna-example-addhit.p4.spec --arch pna --bf-rt-schema ./tmpqfb7g5hb/pna-example-addhit.p4.bfrt.json /tmp/p4c/testdata/p4_16_samples/pna-example-addhit.p4
Error compiling
pna-example-addhit.p4(143): [--Werror=unexpected] error: add_entry add-on-miss not configured or not called from default action: next_hop:
        add_entry(action_name="next_hop", action_params = tmp, expire_time_profile_id = user_meta.timeout);
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

hemant@lower:/tmp/p4c/build$ git status
On branch tighten-addent
Your branch is up to date with 'origin/tighten-addent'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   ../backends/dpdk/dpdkArch.cpp
```

Reply:
I do not know why `-f` is not copying the `.p4-error` output file.  Perhaps it is related to the fact that p4-dpdk back end uses a different suffix `.p4-error` vs. the `.p4-stderr` used by other back ends?  In any case, looks like the quickest workaround is to create the `.p4-error` file you need by copying and pasting the error messages you see in the output to that file.

Reply:
I downloaded latest p4c even in this repo the `-stderr` file has size zero. Also interesting is that it is not a `-error ` file. My issue is I do not see any failed test case added by PNA in p4_16_errors directory and a failed test in p4_16_samples has zero content.

```shell
hemant@lower:/tmp/p4c/testdata$ find -name "pna-add-on-miss-err*"
./p4_16_samples/pna-add-on-miss-err.p4
./p4_16_samples_outputs/pna-add-on-miss-err.p4-stderr
./p4_16_samples_outputs/pna-add-on-miss-err-frontend.p4
./p4_16_samples_outputs/pna-add-on-miss-err-midend.p4
./p4_16_samples_outputs/pna-add-on-miss-err-first.p4
./p4_16_samples_outputs/pna-add-on-miss-err.p4
hemant@lower:/tmp/p4c/testdata$ ls -l ./p4_16_samples_outputs/pna-add-on-miss-err.p4-stderr
-rw-rw-r-- 1 hemant hemant 0 Oct 25 16:53 ./p4_16_samples_outputs/pna-add-on-miss-err.p4-stderr
hemant@lower:/tmp/p4c/testdata$ 
```

Reply:
I am unlikely to give these code changes an approval without explicit test programs that cause the error cases to be exercised.  If you can include or attach in a comment such test programs that you believe should give error messages, and what error messages they produce with your p4c changes, I may have some time to try the code in your PR out on my own system.

Reply:
err1 says "either add_entry is called from a non default_action or add-on-miss is not configured". 
err2 says "add_entry has first parameter as default action".

Test programs to use are:

dpdk/testdata/p4_16_samples/pna-example-addhit.p4.test     // add_entry called from hit action, err1
dpdk/testdata/p4_16_samples/pna-example-miss.p4.test       // add_entry uses default_action for first parameter, err2
dpdk/testdata/p4_16_samples/pna-add-on-miss-err.p4.test   // add on miss is not configured., err2